packages/components/src/components/teaser/teaser.stories.ts

```diff
@@ -0,0 +1,352 @@
+import '../../solid-components';
+import { html } from 'lit-html';
+import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
+
+const { argTypes, parameters } = storybookDefaults('sd-teaser');
+const { overrideArgs } = storybookHelpers('sd-teaser');
+const { generateTemplate } = storybookTemplate('sd-teaser');
+
+export default {
+  title: 'Components/sd-teaser',
+  component: 'sd-teaser',
+  args: overrideArgs([
+    {
+      type: 'slot',
+      name: 'default',
+      value: `<slot-comp style="--slot-height: 48px; --slot-content: 'Main slot'"></slot-comp>`
+    },
+    {
+      type: 'slot',
+      name: 'media',
+      value: `<slot-comp slot='media' style="--slot-height: 48px; --slot-content: 'Media slot'"></slot-comp>`
+    },
+    {
+      type: 'slot',
+      name: 'headline',
+      value: '<h2 slot="headline">Remember: Always Use &lt;h&gt; Tags here!</h2>'
+    },
+    {
+      type: 'slot',
+      name: 'meta',
+      value: `<slot-comp slot='meta' style="--slot-height: 46px; --slot-content: 'Meta slot'"></slot-comp>`
+    }
+  ]),
+  argTypes,
+  parameters: { ...parameters }
+};
+
+/**
+ * This shows sd-teaser in its default state.
+ */
+
+export const Default = {
+  render: (args: any) => {
+    return generateTemplate({ args });
+  }
+};
+
+/**
+ * Teaser in all possible variants.
+ */
+
+export const Variants = {
+  parameters: { controls: { exclude: 'variant' } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'variant' }
+      },
+      args,
+      constants: [
+        {
+          type: 'template',
+          name: 'style',
+          value: '<div style="margin-bottom: 50px">%TEMPLATE%</div>'
+        },
+        {
+          type: 'attribute',
+          name: 'orientation',
+          value: 'vertical'
+        }
+      ]
+    });
+  }
+};
+
+/**
+ * Teaser in all possible combinations of `inset` and 'orientation'. Note that in case the variant is `white border-neutral-300`, the inset is always `true`.
```

Comment:
I would write this then at `VariantAndInset` as this belongs to that.

Furthermore:

![CleanShot 2023-07-27 at 17 16 01@2x](https://github.com/solid-design-system/solid/assets/26542182/423b51a6-7322-4a22-97e6-2baeab4b0cb7)

The elements should all have the same width and height, so that we can compare how they behave differently in regard to their properties. :)